tests/test_bake_project.py

```diff
@@ -0,0 +1,302 @@
+                    "__experimentalist_type": raw
+                    }
+        if subtype == 'custom':
+            d_inputs['custom_experimentalist_type'] = 'new-type'
+
+        with bake_in_temp_dir(cookies, extra_context=d_inputs) as result:
+            assert check_construct(result)
+
+            # Check source code tree structure
+            basename = os.path.basename(result.project)
+            l_tree = tree_list(result.project / 'src')
+            l_tree = [s.split(basename)[1] for s in l_tree]
+
+            # Add to dict
+            d_subtypes[subtype] = {'basename': basename, 'tree': l_tree}
+
+    # Check directory names
+    assert d_subtypes['sampler'][
+               'basename'] == 'autora-experimentalist-sampler-test-experimentalist'
+    assert d_subtypes['pooler'][
+               'basename'] == 'autora-experimentalist-pooler-test-experimentalist'
+    assert d_subtypes['custom'][
+               'basename'] == 'autora-experimentalist-new-type-test-experimentalist'
+
+    # Check directory trees
+    assert d_subtypes['sampler']['tree'] == ['/src/autora', '/src/autora/experimentalist',
+                                             '/src/autora/experimentalist/sampler',
+                                             '/src/autora/experimentalist/sampler/test_experimentalist',
+                                             '/src/autora/experimentalist/sampler/test_experimentalist/__init__.py']
+
+    assert d_subtypes['pooler']['tree'] == ['/src/autora', '/src/autora/experimentalist',
+                                            '/src/autora/experimentalist/pooler',
+                                            '/src/autora/experimentalist/pooler/test_experimentalist',
+                                            '/src/autora/experimentalist/pooler/test_experimentalist/__init__.py']
+
+    assert d_subtypes['custom']['tree'] == ['/src/autora', '/src/autora/experimentalist',
+                                            '/src/autora/experimentalist/new_type',
+                                            '/src/autora/experimentalist/new_type/test_experimentalist',
+                                            '/src/autora/experimentalist/new_type/test_experimentalist/__init__.py']
+
+
+def test_runner(cookies):
+    subtypes_raw = [
+        "{% if cookiecutter.autora_contribution_type == 'experiment_runner' -%}experiment_runner{% else -%}N/A - Press Enter to Skip{% endif -%}",
+        "{% if cookiecutter.autora_contribution_type == 'experiment_runner' -%}experimentation_manager{% else -%}N/A - Press Enter to Skip{% endif -%}",
+        "{% if cookiecutter.autora_contribution_type == 'experiment_runner' -%}recruitment_manager{% else -%}N/A - Press Enter to Skip{% endif -%}"]
+    subtypes = ['experiment_runner', 'experimentation_manager', 'recruitment_manager']
+
+    d_subtypes = {}
+    for i, (subtype, raw) in enumerate(zip(subtypes, subtypes_raw)):
+        d_inputs = {"contribution_name": "test_runner",
```

Comment:
i don't think any material hinges on it, but just wondering about the difference in naming conventions in using underscore vs dash. it looks like we have `contribution_names` "`test-theorist`" and "`test-experimentalist`" with dashes but "`test_runner`" and "`test_custom`" with underscores. should they all be using underscores?

Reply:
I think this should be test-runner and test-custom. It is very confusing because of the different naming conventions in the package naming and the folder structure. Thanks for paying so close attention!

Reply:
i also like the differentiation between hierarchy (dashes) and compound names (underscores). so other instances of case 1 would be something like:

_Theorist_
**contribution name:** `test_theorist`
**project name:** autora-theorist-test_theorist
**modules:**  test_theorist

_Experimentalist_
**contribution name:** `test_experimentalist`
**project name:** autora-experimentalist-test_experimentalist
**modules:**  test_experimentalist

_Experimentation Manager_
**contribution name:** `test_experimentation_manager`
**project name:** autora-experimentation_manager-test_experimentation_manager
**modules:**  test_experimentation_manager

_Synthetic Data_
**contribution name:** `test_synthetic_data`
**project name:** autora-synthetic_data-test_synthetic_data
**modules:**  test_synthetic_data